src/repositories/posts.repository.js

```diff
@@ -58,15 +58,23 @@ exports.create = async (newPost, result) => {
       }
     }
 
-    for (const item of mapAllTagsWithoutDesc) {
-      const tagDescription = await investApi.fetchTagDesc(item);
+    let mapAllTagsWithoutDescString = '';
 
-      mapNewTags.push({
-        tagname: item,
-        description: tagDescription,
-      });
+    /**
+     * prepare a string of tags with ";" as delimeter
+     * for eg:- [java, javascript] will become "java;javascript"
+     */
+    for (let i = 0; i < mapAllTagsWithoutDesc.length; i++) {
+      if (i === mapAllTagsWithoutDesc.length - 1) {
+        mapAllTagsWithoutDescString += `${mapAllTagsWithoutDesc[i]}`
+      } else {
+        mapAllTagsWithoutDescString += `${mapAllTagsWithoutDesc[i]};`
+      }
     }
 
+    const resp = await investApi.fetchTagDesc(mapAllTagsWithoutDescString)
+    mapNewTags = investApi.prepareTags(mapAllTagsWithoutDesc, resp)
+
```

Comment:
Why not just call `prepareTags` inside `fetchTagDesc` and return the already prepared response?

Reply:
It would be better to control the flow from the main function rather than calling functions inside function, it would become a whole task in itself to understand who is calling what, also it would be easier to test a function if it only has single responsibility.